src/charm.py

```diff
@@ -48,18 +74,34 @@ def _on_config_changed(self, _):
         self._stored.listen_port = self.config.get("exporter-listen-port")
         self._stored.stats_groups = self.config.get("exporter-stats-groups")
         self._manage_prometheus_bind_exporter_service()
+
+        # update relation data
+        bind_exporter_relation = self.model.get_relation("bind-exporter")
+        if bind_exporter_relation:
+            logger.info("Updating `bind-exporter` relation data.")
+            bind_exporter_relation.data[self.unit].update({
+                "hostname": self.private_address, "port": str(self._stored.listen_port)
+            })
+
         self.unit.status = ActiveStatus("Unit is ready")
 
-    def _manage_prometheus_bind_exporter_service(self):
-        """Manage the prometheus-bind-exporter service."""
-        logger.debug("prometheus-bind-exporter configuration in progress")
-        private_address = self.model.get_binding("designate-bind").network.bind_address
-        subprocess.check_call([
-            "snap", "set", "prometheus-bind-exporter",
-            f"web.listen-address={private_address or ''}:{self._stored.listen_port}",
-            f"web.stats-groups={self._stored.stats_groups}"
-        ])
-        logger.info("prometheus-bind-exporter has been reconfigured")
+    def _on_bind_exporter_relation_changed(self, event: RelationChangedEvent):
+        """Prometheus relation changed hook.
+
+        This hook will ensure the creation of a new target in Prometheus.
+        """
+        logger.info("Registering new `%s` target in Prometheus.", self.unit.name)
```

Comment:
I think it would be more accurate to say "Shared relation data with self.unit.name".